Compilation error on macOS Sequoia 15.4, outdated zlib

I'm hitting a compilation error that I think may be related to the outdated version of zlib.

Furthermore, it looks like at least some of the documentation in this repo says zlib version 1.2.12 is used, wheras the zlib source included here says that it is version 1.2.13.

The current version of zlib is 1.3.1, released January 22, 2024.

```
xmp-toolkit-rs on  main is 📦 v1.9.2 via 🦀 v1.86.0
❯ cargo build
    Updating crates.io index
     Locking 2 packages to latest compatible versions
    Updating thiserror v1.0.69 -> v2.0.12
    Updating thiserror-impl v1.0.69 -> v2.0.12
   Compiling xmp_toolkit v1.9.2 (/Users/schell/code/xmp-toolkit-rs)
warning: xmp_toolkit@1.9.2: In file included from /Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out/external/xmp_toolkit/third-party/zlib/zutil.c:10:
warning: xmp_toolkit@1.9.2: In file included from /Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out/external/xmp_toolkit/third-party/zlib/gzguts.h:21:
warning: xmp_toolkit@1.9.2: In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/stdio.h:108:
warning: xmp_toolkit@1.9.2: In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stdio.h:61:
warning: xmp_toolkit@1.9.2: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_stdio.h:318:7: error: expected unqualified-id
warning: xmp_toolkit@1.9.2:   318 | FILE    *fdopen(int, const char *) __DARWIN_ALIAS_STARTING(__MAC_10_6, __IPHONE_2_0, __DARWIN_ALIAS(fdopen));
warning: xmp_toolkit@1.9.2:       |          ^
warning: xmp_toolkit@1.9.2: /Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out/external/xmp_toolkit/third-party/zlib/zutil.h:147:33: note: expanded from macro 'fdopen'
warning: xmp_toolkit@1.9.2:   147 | #        define fdopen(fd,mode) NULL /* No fdopen() */
warning: xmp_toolkit@1.9.2:       |                                 ^
warning: xmp_toolkit@1.9.2: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/17/include/__stddef_null.h:21:14: note: expanded from macro 'NULL'
warning: xmp_toolkit@1.9.2:    21 | #define NULL __null
warning: xmp_toolkit@1.9.2:       |              ^
warning: xmp_toolkit@1.9.2: 1 error generated.
warning: xmp_toolkit@1.9.2: ToolExecError: command did not execute successfully (status code exit status: 1): env -u IPHONEOS_DEPLOYMENT_TARGET LC_ALL="C" "c++" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "--target=arm64-apple-macosx" "-mmacosx-version-min=15.4" "-I" "external/xmp_toolkit/XMPCore/resource/mac" "-I" "external/xmp_toolkit/XMPFiles/resource/mac" "-I" "/Users/schell/code/xmp-toolkit-rs/external/xmp_toolkit" "-I" "/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out/external/xmp_toolkit" "-I" "external/xmp_toolkit/build" "-I" "external/xmp_toolkit/public/include" "-I" "external/xmp_toolkit/XMPFilesPlugins/api/source" "-Wall" "-Wextra" "-Wno-bitwise-instead-of-logical" "-Wno-deprecated-declarations" "-Wno-deprecated-register" "-Wno-int-in-bool-context" "-Wno-macro-redefined" "-Wno-null-conversion" "-Wno-unused-but-set-variable" "-std=c++17" "-Wno-deprecated" "-Wno-deprecated-declarations" "-Wno-missing-field-initializers" "-Wno-reorder" "-Wno-unused-function" "-Wno-unused-parameter" "-Wno-unused-variable" "-Wnon-virtual-dtor" "-Woverloaded-virtual" "-DMAC_ENV=1" "-DXMP_MacBuild=1" "-D_LARGEFILE64_SOURCE" "-DXML_DEV_URANDOM" "-DXMP_OSX_SDK=13.1" "-DXMP_OSX_TARGET=10.15" "-DAPPLE_UNIVERSAL=1" "-DTXMP_STRING_TYPE=std::string" "-DXML_STATIC=1" "-DXMP_StaticBuild=1" "-DHAVE_EXPAT_CONFIG_H=1" "-o" "/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out/1b9f9092d09ed7a9-zutil.o" "-c" "/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out/external/xmp_toolkit/third-party/zlib/zutil.c"
error: failed to run custom build command for `xmp_toolkit v1.9.2 (/Users/schell/code/xmp-toolkit-rs)`

Caused by:
  process didn't exit successfully: `/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-ad7fcdc9f08fb3b6/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-changed=build.rs
  > git submodule init

  --- stdout ---


  --- stderr ---



  > git submodule update

  --- stdout ---


  --- stderr ---



  cargo:info=COPYING /Users/schell/code/xmp-toolkit-rs/external/libexpat/expat/lib to /Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out/external/xmp_toolkit/third-party/expat
  Copying zlib to third_party dir ...
  cargo:info=COPYING /Users/schell/code/xmp-toolkit-rs/external/zlib to /Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out/external/xmp_toolkit/third-party
  cargo:rustc-link-lib=framework=Carbon
  cargo:rustc-link-lib=framework=Security
  exit status: 0
  exit status: 0
  exit status: 0
  cargo:rustc-link-search=native=/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out
  cargo:include=/Users/schell/code/xmp-toolkit-rs/external/xmp_toolkit/public/include
  OUT_DIR = Some(/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out)
  OPT_LEVEL = Some(0)
  TARGET = Some(aarch64-apple-darwin)
  HOST = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=CXX_aarch64-apple-darwin
  CXX_aarch64-apple-darwin = None
  cargo:rerun-if-env-changed=CXX_aarch64_apple_darwin
  CXX_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=HOST_CXX
  HOST_CXX = None
  cargo:rerun-if-env-changed=CXX
  CXX = None
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  RUSTC_WRAPPER = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some(true)
  cargo:rerun-if-env-changed=MACOSX_DEPLOYMENT_TARGET
  MACOSX_DEPLOYMENT_TARGET = None
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64_apple_darwin
  CXXFLAGS_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64-apple-darwin
  CXXFLAGS_aarch64-apple-darwin = None
  CARGO_ENCODED_RUSTFLAGS = Some()
  OUT_DIR = Some(/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out)
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  TARGET = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=MACOSX_DEPLOYMENT_TARGET
  MACOSX_DEPLOYMENT_TARGET = None
  HOST = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64_apple_darwin
  CXXFLAGS_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64-apple-darwin
  CXXFLAGS_aarch64-apple-darwin = None
  OUT_DIR = Some(/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out)
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  TARGET = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=MACOSX_DEPLOYMENT_TARGET
  MACOSX_DEPLOYMENT_TARGET = None
  HOST = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64_apple_darwin
  CXXFLAGS_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64-apple-darwin
  CXXFLAGS_aarch64-apple-darwin = None
  OUT_DIR = Some(/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out)
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  TARGET = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=MACOSX_DEPLOYMENT_TARGET
  MACOSX_DEPLOYMENT_TARGET = None
  HOST = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64_apple_darwin
  CXXFLAGS_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64-apple-darwin
  CXXFLAGS_aarch64-apple-darwin = None
  OUT_DIR = Some(/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out)
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  TARGET = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=MACOSX_DEPLOYMENT_TARGET
  MACOSX_DEPLOYMENT_TARGET = None
  HOST = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64_apple_darwin
  CXXFLAGS_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64-apple-darwin
  CXXFLAGS_aarch64-apple-darwin = None
  OUT_DIR = Some(/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out)
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  TARGET = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=MACOSX_DEPLOYMENT_TARGET
  MACOSX_DEPLOYMENT_TARGET = None
  HOST = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64_apple_darwin
  CXXFLAGS_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64-apple-darwin
  CXXFLAGS_aarch64-apple-darwin = None
  OUT_DIR = Some(/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out)
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  TARGET = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=MACOSX_DEPLOYMENT_TARGET
  MACOSX_DEPLOYMENT_TARGET = None
  HOST = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64_apple_darwin
  CXXFLAGS_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64-apple-darwin
  CXXFLAGS_aarch64-apple-darwin = None
  OUT_DIR = Some(/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out)
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  TARGET = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=MACOSX_DEPLOYMENT_TARGET
  MACOSX_DEPLOYMENT_TARGET = None
  HOST = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64_apple_darwin
  CXXFLAGS_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64-apple-darwin
  CXXFLAGS_aarch64-apple-darwin = None
  OUT_DIR = Some(/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out)
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  TARGET = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=MACOSX_DEPLOYMENT_TARGET
  MACOSX_DEPLOYMENT_TARGET = None
  HOST = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64_apple_darwin
  CXXFLAGS_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64-apple-darwin
  CXXFLAGS_aarch64-apple-darwin = None
  OUT_DIR = Some(/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out)
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  TARGET = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=MACOSX_DEPLOYMENT_TARGET
  MACOSX_DEPLOYMENT_TARGET = None
  HOST = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64_apple_darwin
  CXXFLAGS_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64-apple-darwin
  CXXFLAGS_aarch64-apple-darwin = None
  OUT_DIR = Some(/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out)
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  TARGET = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=MACOSX_DEPLOYMENT_TARGET
  MACOSX_DEPLOYMENT_TARGET = None
  HOST = Some(aarch64-apple-darwin)
  cargo:rerun-if-env-changed=CXXFLAGS
  CXXFLAGS = None
  cargo:rerun-if-env-changed=HOST_CXXFLAGS
  HOST_CXXFLAGS = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64_apple_darwin
  CXXFLAGS_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=CXXFLAGS_aarch64-apple-darwin
  CXXFLAGS_aarch64-apple-darwin = None
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  cargo:warning=In file included from /Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out/external/xmp_toolkit/third-party/zlib/zutil.c:10:
  cargo:warning=In file included from /Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out/external/xmp_toolkit/third-party/zlib/gzguts.h:21:
  cargo:warning=In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/stdio.h:108:
  cargo:warning=In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stdio.h:61:
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/_stdio.h:318:7: error: expected unqualified-id
  cargo:warning=  318 | FILE    *fdopen(int, const char *) __DARWIN_ALIAS_STARTING(__MAC_10_6, __IPHONE_2_0, __DARWIN_ALIAS(fdopen));
  cargo:warning=      |          ^
  cargo:warning=/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out/external/xmp_toolkit/third-party/zlib/zutil.h:147:33: note: expanded from macro 'fdopen'
  cargo:warning=  147 | #        define fdopen(fd,mode) NULL /* No fdopen() */
  cargo:warning=      |                                 ^
  cargo:warning=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/17/include/__stddef_null.h:21:14: note: expanded from macro 'NULL'
  cargo:warning=   21 | #define NULL __null
  cargo:warning=      |              ^
  cargo:warning=1 error generated.
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 0
  exit status: 1
  cargo:warning=ToolExecError: command did not execute successfully (status code exit status: 1): env -u IPHONEOS_DEPLOYMENT_TARGET LC_ALL="C" "c++" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "--target=arm64-apple-macosx" "-mmacosx-version-min=15.4" "-I" "external/xmp_toolkit/XMPCore/resource/mac" "-I" "external/xmp_toolkit/XMPFiles/resource/mac" "-I" "/Users/schell/code/xmp-toolkit-rs/external/xmp_toolkit" "-I" "/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out/external/xmp_toolkit" "-I" "external/xmp_toolkit/build" "-I" "external/xmp_toolkit/public/include" "-I" "external/xmp_toolkit/XMPFilesPlugins/api/source" "-Wall" "-Wextra" "-Wno-bitwise-instead-of-logical" "-Wno-deprecated-declarations" "-Wno-deprecated-register" "-Wno-int-in-bool-context" "-Wno-macro-redefined" "-Wno-null-conversion" "-Wno-unused-but-set-variable" "-std=c++17" "-Wno-deprecated" "-Wno-deprecated-declarations" "-Wno-missing-field-initializers" "-Wno-reorder" "-Wno-unused-function" "-Wno-unused-parameter" "-Wno-unused-variable" "-Wnon-virtual-dtor" "-Woverloaded-virtual" "-DMAC_ENV=1" "-DXMP_MacBuild=1" "-D_LARGEFILE64_SOURCE" "-DXML_DEV_URANDOM" "-DXMP_OSX_SDK=13.1" "-DXMP_OSX_TARGET=10.15" "-DAPPLE_UNIVERSAL=1" "-DTXMP_STRING_TYPE=std::string" "-DXML_STATIC=1" "-DXMP_StaticBuild=1" "-DHAVE_EXPAT_CONFIG_H=1" "-o" "/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out/1b9f9092d09ed7a9-zutil.o" "-c" "/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out/external/xmp_toolkit/third-party/zlib/zutil.c"
  exit status: 0
  exit status: 0

  --- stderr
  INFO: building standard FFI for crate
  [build.rs:358:5] &src_path = "/Users/schell/code/xmp-toolkit-rs/external/libexpat/expat/lib"
  [build.rs:358:5] &src_path = "/Users/schell/code/xmp-toolkit-rs/external/zlib"


  error occurred in cc-rs: command did not execute successfully (status code exit status: 1): env -u IPHONEOS_DEPLOYMENT_TARGET LC_ALL="C" "c++" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-2" "-fno-omit-frame-pointer" "--target=arm64-apple-macosx" "-mmacosx-version-min=15.4" "-I" "external/xmp_toolkit/XMPCore/resource/mac" "-I" "external/xmp_toolkit/XMPFiles/resource/mac" "-I" "/Users/schell/code/xmp-toolkit-rs/external/xmp_toolkit" "-I" "/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out/external/xmp_toolkit" "-I" "external/xmp_toolkit/build" "-I" "external/xmp_toolkit/public/include" "-I" "external/xmp_toolkit/XMPFilesPlugins/api/source" "-Wall" "-Wextra" "-Wno-bitwise-instead-of-logical" "-Wno-deprecated-declarations" "-Wno-deprecated-register" "-Wno-int-in-bool-context" "-Wno-macro-redefined" "-Wno-null-conversion" "-Wno-unused-but-set-variable" "-std=c++17" "-Wno-deprecated" "-Wno-deprecated-declarations" "-Wno-missing-field-initializers" "-Wno-reorder" "-Wno-unused-function" "-Wno-unused-parameter" "-Wno-unused-variable" "-Wnon-virtual-dtor" "-Woverloaded-virtual" "-DMAC_ENV=1" "-DXMP_MacBuild=1" "-D_LARGEFILE64_SOURCE" "-DXML_DEV_URANDOM" "-DXMP_OSX_SDK=13.1" "-DXMP_OSX_TARGET=10.15" "-DAPPLE_UNIVERSAL=1" "-DTXMP_STRING_TYPE=std::string" "-DXML_STATIC=1" "-DXMP_StaticBuild=1" "-DHAVE_EXPAT_CONFIG_H=1" "-o" "/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out/1b9f9092d09ed7a9-zutil.o" "-c" "/Users/schell/code/xmp-toolkit-rs/target/debug/build/xmp_toolkit-fe90a55d5b8c1dbd/out/external/xmp_toolkit/third-party/zlib/zutil.c"
```

--- a/external/zlib/zutil.h
+++ b/external/zlib/zutil.h
@@ -144,7 +144,7 @@ extern z_const char * const z_errmsg[10]; /* indexed by 2-zlib_error */
 #      include <unix.h> /* for fdopen */
 #    else
 #      ifndef fdopen
-#        define fdopen(fd,mode) NULL /* No fdopen() */
+// #        define fdopen(fd,mode) NULL /* No fdopen() */
 #      endif
 #    endif
 #  endif

